src/vs/platform/workspace/common/editSessions.ts

```diff
@@ -0,0 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken, CancellationTokenSource } from 'vs/base/common/cancellation';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { IWorkspaceFolder } from 'vs/platform/workspace/common/workspace';
+
+export interface IEditSessionIdentityProvider {
```

Comment:
I think going forward I would call "edit sessions" rather "workspace session" because I may still have a session even though I am not editing a workspace.

Reply:
Tracking this feedback in https://github.com/microsoft/vscode/issues/157734#issuecomment-1219698111--we should revisit the name if/when we implement additional consumers of this proposal.